osu.Game.Rulesets.Catch/Judgements/CatchBananaJudgement.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Game.Rulesets.Scoring;
+
+namespace osu.Game.Rulesets.Catch.Judgements
+{
+    public class CatchBananaJudgement : CatchJudgement
+    {
+        public override bool AffectsCombo => false;
+        public override bool IsBonus => true;
```

Comment:
Instead of adding cases onto this stuff, let's re-order `Judgement` into the following:

```
class Judgement:
    virtual ScoreType ScoreType => ScoreType.Combo
    virtual bool IncreasesCombo => ScoreType == ScoreType.Combo

enum ScoreType:
    Combo
    Bonus
```

Reply:
Well the discussion has changed a bit since the inception of this PR, so this functionality may not be needed in general. I don't know what juice stream droplets need to be counted as to continue this discussion further.